x-pack/legacy/plugins/apm/public/components/shared/Links/apm/ExternalLinks.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import url from 'url';
+import { IUrlParams } from '../../../../context/UrlParamsContext/types';
+
+export type QueryParams = Required<Pick<IUrlParams, 'rangeFrom' | 'rangeTo'>>;
+
+export const getTraceUrl = (
+  traceId: number | string,
+  queryParams?: QueryParams
+) => {
+  const { rangeFrom, rangeTo } = queryParams || {};
+  return url.format({
+    pathname: `/link-to/trace/${traceId}`,
+    query: rangeFrom && rangeTo ? { rangeFrom, rangeTo } : {}
+  });
+};
```

Comment:
I don't think it's great to mix positional and named arguments (there are use cases, like a single required arg, and a host of optional ones) but in this case all args are required.

I don't think `traceId` can be both a string and a number. It's a string in elasticsearch, right? 

```ts
export const getTraceUrl = ({
  traceId,
  rangeFrom,
  rangeTo
}: {
  traceId: string;
  rangeFrom: string;
  rangeTo: string;
}) => {
  return url.format({
    pathname: `/link-to/trace/${traceId}`,
    query: { rangeFrom, rangeTo }
  });
};
```

Reply:
And I don't think we need 
```
export type QueryParams = Required<Pick<IUrlParams, 'rangeFrom' | 'rangeTo'>>;
```

Reply:
Yes, it is